chemprop/models/model.py

```diff
@@ -93,62 +101,65 @@ def create_ffn(self, args: TrainArgs) -> None:
+        else:
+            atom_first_linear_dim = first_linear_dim
 
-        dropout = nn.Dropout(args.dropout)
-        activation = get_activation_function(args.activation)
+        if args.bond_descriptors == 'descriptor':
+            bond_first_linear_dim = first_linear_dim + args.bond_descriptors_size
+        else:
+            bond_first_linear_dim = first_linear_dim
 
         # Create FFN layers
-        if args.ffn_num_layers == 1:
-            ffn = [
-                dropout,
-                nn.Linear(first_linear_dim, self.output_size)
-            ]
+        if self.is_atom_bond_targets:
+            self.readout = MultiReadout(atom_features_size=atom_first_linear_dim,
+                                        bond_features_size=bond_first_linear_dim,
+                                        atom_hidden_size=args.ffn_hidden_size + args.atom_descriptors_size,
+                                        bond_hidden_size=args.ffn_hidden_size + args.bond_descriptors_size,
+                                        num_layers=args.ffn_num_layers,
+                                        output_size=self.relative_output_size,
+                                        dropout=args.dropout,
+                                        activation=args.activation,
+                                        atom_constraints=args.atom_constraints,
+                                        bond_constraints=args.bond_constraints,
+                                        shared_ffn=args.shared_atom_bond_ffn,
+                                        weights_ffn_num_layers=args.weights_ffn_num_layers)
         else:
-            ffn = [
-                dropout,
-                nn.Linear(first_linear_dim, args.ffn_hidden_size)
-            ]
-            for _ in range(args.ffn_num_layers - 2):
-                ffn.extend([
-                    activation,
-                    dropout,
-                    nn.Linear(args.ffn_hidden_size, args.ffn_hidden_size),
-                ])
-            ffn.extend([
-                activation,
-                dropout,
-                nn.Linear(args.ffn_hidden_size, self.output_size),
-            ])
-
-        # If spectra model, also include spectra activation
-        if args.dataset_type == 'spectra':
-            if args.spectra_activation == 'softplus':
-                spectra_activation = nn.Softplus()
-            else:  # default exponential activation which must be made into a custom nn module
-                class nn_exp(torch.nn.Module):
-                    def __init__(self):
-                        super(nn_exp, self).__init__()
-
-                    def forward(self, x):
-                        return torch.exp(x)
-
-                spectra_activation = nn_exp()
-            ffn.append(spectra_activation)
-
-        # Create FFN model
-        self.ffn = nn.Sequential(*ffn)
+            self.readout = build_ffn(first_linear_dim=atom_first_linear_dim,
```

Comment:
was this change in the model submodule name intentional? I don't disagree with the naming (I think it's more logically correct), but it will *break* module saving/loading for older versions of chemprop models because torch internals load modules based on key names

Reply:
The backward compatibility for parameter names is taken care in chemprop/utils.py.

Reply:
that seems like overkill to change a module name when the change isn't strictly necessary in the first place. Also, something that would need to be discussed with the dev team at large.

Reply:
I totally agree. How about still sticking with the original name here?